From 88da398b39b2047f3f42e56d3d3a24ba18233e40 Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Tue, 22 Sep 2009 17:26:06 +0200
Subject: [PATCH] Don't rely on activation_uri when detecting mounts

We test for the activation_uri (i.e. G_FILE_ATTRIBUTE_STANDARD_TARGET_URI)
which is usually not exposed from computer:// items until the item is mounted
(and that's up to particular gvfs volume monitor).

Simply don't test for activation_uri, file type of G_FILE_TYPE_MOUNTABLE
should be enough.
---
 libnautilus-private/nautilus-directory-async.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/libnautilus-private/nautilus-directory-async.c b/libnautilus-private/nautilus-directory-async.c
index 653fd30..ff20ac5 100644
--- a/libnautilus-private/nautilus-directory-async.c
+++ b/libnautilus-private/nautilus-directory-async.c
@@ -1700,9 +1700,8 @@ lacks_mount (NautilusFile *file)
 		 (file->details->type == G_FILE_TYPE_DIRECTORY &&
 		  nautilus_file_is_self_owned (file)) ||
 		 
-		 /* Mountable with a target_uri, could be a mountpoint */
-		 (file->details->type == G_FILE_TYPE_MOUNTABLE &&
-		  file->details->activation_uri != NULL)
+		 /* Mountable, could be a mountpoint */
+		 (file->details->type == G_FILE_TYPE_MOUNTABLE)
 
 		 )
 		);
-- 
1.6.4.4


